biojava-structure/src/main/java/org/biojava/nbio/structure/basepairs/TertiaryBasePairParameters.java

```diff
@@ -0,0 +1,136 @@
+/*
+ *                    BioJava development code
+ *
+ * This code may be freely distributed and modified under the
+ * terms of the GNU Lesser General Public Licence.  This should
+ * be distributed with the code.  If you do not have a copy,
+ * see:
+ *
+ *      http://www.gnu.org/copyleft/lesser.html
+ *
+ * Copyright for this code is held jointly by the individual
+ * authors.  These should be listed in @author doc comments.
+ *
+ * For more information on the BioJava project and its aims,
+ * or to join the biojava-l mailing list, visit the home page
+ * at:
+ *
+ *      http://www.biojava.org/
+ *
+ *
+ */
+package org.biojava.nbio.structure.basepairs;
+
+import org.biojava.nbio.structure.Atom;
+import org.biojava.nbio.structure.Chain;
+import org.biojava.nbio.structure.Group;
+import org.biojava.nbio.structure.Structure;
+import org.biojava.nbio.structure.contact.Pair;
+
+import javax.vecmath.Matrix4d;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class also finds the base pairing and base-pair step parameters but has a broader definition
+ * of a base pair so that non-canonical-WC base pairs will be detected and reported.  This is useful
+ * for RNA that has folded into different regions, and for higher-order DNA structures.  Intra-strand
+ * pairings are considered in this class (but in not the base class or MismatchedBasePairParameters class)
+ * @author Luke Czapla
+ * @since 5.0.0
+ *
+ */
+public class TertiaryBasePairParameters extends BasePairParameters implements Serializable {
+
+    // These are the criteria used to select proper base pairs.
+    protected static double MaxStagger = 2.0, MaxPropeller = 60.0;
```

Comment:
I agree with you there.  So I made these values non-static (for thread safety with multiple independent instances) and created the final DEFAULT_MAX_STAGGER etc.  I'm not sure the use of the final keyword in the setter, it's possible that someone would create the object, call analyze() once and look at that data, then change the maxStagger or one of those values, and then call analyze() again and look at the new data.  That would be like if someone wanted to see those weird "base pairs" that Wilma Olson showed me, they were hardly really paired because Stagger is a measure of how planar the pair is.  But there is a lot of distorted crap that specialists might like to see.